great_expectations/expectations/expectation.py

```diff
@@ -2281,6 +2283,27 @@ def _get_final_maturity_level(
 
         return maturity
 
+    def register_save_callback(
+        self, callback: Callable[[Expectation], Expectation]
+    ) -> None:
+        """Provide the Expectation with a callback used to save the Expectation configuration."""
+        self._save_callback = callback
+
+    def transfer_save_callback(self, expectation: Expectation) -> Expectation:
+        """Pass the save callback from this Expectation to another."""
+        expectation.register_save_callback(callback=self._save_callback)
+        return expectation
+
+    @public_api
+    def save(self):
+        """Save this Expectation."""
+        if not self._save_callback:
+            raise RuntimeError(
+                "Expectation must be added to an ExpectationSuite before it can be saved."
+            )
+        expectation = self._save_callback(self)
+        expectation.transfer_save_callback(expectation=self)
```

Comment:
This seems like an odd method to be public. Why not have the transfer happen as part of `_save_callback`?

Reply:
you know, not only are you right, but that's already happening 🙃 i was thinking about how to allow a user to continue editing and saving the same expectation, and solved it twice in two places. Better to just allow the suite to register the new callback. Thanks!